libtenzir/builtins/operators/decapsulate.cpp

```diff
@@ -268,109 +281,110 @@ auto parse(record_ref builder, std::span<const std::byte> bytes,
   return std::nullopt;
 }
 
-struct operator_args {
-  std::optional<located<uint16_t>> vxlan_port;
-
-  friend auto inspect(auto& f, operator_args& x) -> bool {
-    return f.object(x)
-      .pretty_name("operator_args")
-      .fields(f.field("vxlan_port", x.vxlan_port));
+auto decapsulate(series s, diagnostic_handler& dh) -> std::optional<series> {
+  if (s.type.name() != "pcap.packet") {
+    diagnostic::warning("cannot decapsulate schema '{}'", s.type.name())
```

Comment:
I know that this check was also present in the original, but we probably don't want to have this anymore (in particular because it poorly interacts with functions, because the inner type would have to have a name, which is currently only possible by defining a custom schema). Instead, if it quacks like a duck (has the necessary fields), then it probably is a duck.